app/models/saved_scenario.rb

```diff
@@ -127,6 +127,17 @@ def update_with_api_params(params)
     incoming_id = params[:scenario_id]
     add_id_to_history(scenario_id) if incoming_id && scenario_id != incoming_id
```

Comment:
I think we can move this validation into the `add_id_to_history` method: that would make the validation available for all other code that updates SavedScenarios. Then you also don't need the extra validation step in the service anymore.
Would you mind adding a short test for it as well? Thanks!